protocols/kad/src/behaviour.rs

```diff
@@ -2472,6 +2523,43 @@ pub enum KademliaEvent {
     PendingRoutablePeer { peer: PeerId, address: Multiaddr },
 }
 
+/// Information about progress events.
+#[derive(Debug, Clone)]
+pub struct ProgressStep {
+    /// The index into the event
+    pub count: NonZeroUsize,
+    /// Is this the final event?
+    pub last: bool,
+}
+
+impl ProgressStep {
+    // TODO Why the pub?
+    pub(crate) fn first() -> Self {
```

Comment:
I don't think `pub(crate)` is needed. See 0f0399bae.

Reply:
merged, 🙏